lib/ProgramCall.js

```diff
@@ -51,28 +45,46 @@ class ProgramCall {
       opt = type;
     } else {
       opt = options;
```

Comment:
I think we need to overhaul this interface in a future PR. This interface is quite clunky.

Reply:
I agree